keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1129 @@
+  kind: Pod
+  spec:
+    resources:
+      requests:
+        memory: 100Mi
+        cpu: 2
+      limits:
+        memory: 200Mi
+        cpu: 3
+    initContainers:
+    - name: init-myservice
+      resources:
+        requests:
+          memory: 300Mi
+          cpu: 4
+        limits:
+          memory: 600Mi
+          cpu: 6
+    containers:
+    - name: container-1
+      resources:
+        requests:
+          memory: 50Mi
+          cpu: 1
+        limits:
+          memory: 100Mi
+          cpu: 2
+      - name: container-2
+        resources:
+          requests:
+            memory: 50Mi
+            cpu: 1
+          limits:
+            memory: 100Mi
+            cpu: 2 
+  ```
+
+  **Resources requests at pod-level**
+
+  * The scheduler will use the higher of init and pod-level requests values to find the suitable node for pod placement.
+    
+    From the example above, the scheduler will use:
+
+      max(init container resources requests, pod-level resources requests) 
+
+        = max (memory: 300Mi cpu: 4, memory: 100 Mi cpu: 2)
+
+        = (memory: 300 Mi, cpu: 4) to find a suitable node.
+
+  * The init containers will rely on its own requests specified in the PodSpec. 
+  * After the init containers execution is completed, kubelet will use the pod-level CPU request to set cpu.weight at pod-level cgroup.
+
+  **Resources limits at pod-level**
+
+  * The init containers will rely on their own limits specified in the PodSpec.
+  * After the init containers execution is completed, kubelet will use CPU and memory limits at pod-level, and translate them to cpu.max and memory.max cgroup settings
+  at pod level respectively. 
+
+  **Pros**
+
+  * Reduces the problem of overprovisioning.
+
+  **Cons**
+
+  * Complicates the definitions of QoS classes as this will require considering requests/limits for both initContainer and pod-level requests/limits to attach a 
+  QoS class with a pod. It might be worth setting dynamic QoS classes if this option were to be implemented. The QoS class in the initialization phase is based on
+  init containers requests/limits, while the QoS class after the initialization phase takes pod-level requests/limits into account. 
+  * Complicated to implement as it requires additional logic in kubelet to monitor init containers completion after which it will set/change pod level cgroup settings.
+
+**Note: Option 1 is a *preferred* design option as it is easier to implement, and also easier to understand from UX perspective. 
+If there are enough supporting use cases for Option 2, we can explore the implementation in the beta phase of the KEP. Option 2
+will also require discussions about dynamic QoS classes as per the execution phase of the pod (init vs non-init phase).**
+
+
+#### Admission Controller
+
+* LimitRanger
+
+  This admission controller applies default resource requests to the Pods. In the existing implementation (without this KEP), it sums up container-level requests to calculate
```

Comment:
"default resource requests to the Pods." --> This is applied to individual containers in a pod, right?

Reply:
PTAL, I reworded the entire admission controller section with more refined details.